File: Algo/warmups/Check_GivenSum_inArray.js
B = [1, 2, 7, 9];
//B = [1, 2, 4, 4];
sum = 8;

var hashTable = {};
var matched = false;
for (var i in B) {

    let comp = sum - B[i];
    if (comp in hashTable) {
        matched = true;
        break;
    }
    hashTable[comp] = i;
}

console.log('matched', matched);

File: Algo/hr_warmups/Check_GivenSum_inArray.py
B = [1, 2, 7, 9]
B = [1, 2, 4, 4]
sum = 8

hashTable = []
matched = False
for i in range(len(B)):
	comp = sum - B[i]
	if comp in hashTable:
		matched = True
		break

	hashTable.append(comp)
	print comp, B[i], matched

print 'matched=',matched

